modin/engines/base/io/text/csv_dispatcher.py

```diff
@@ -83,115 +91,141 @@ def _read(cls, filepath_or_buffer, **kwargs):
-                    if len(column_names) > (column_chunksize * (i + 1))
-                    else 0
-                    if len(column_names) < (column_chunksize * i)
-                    else len(column_names) - (column_chunksize * i)
-                    for i in range(num_splits)
-                ]
-
-            args = {
-                "fname": filepath_or_buffer,
-                "num_splits": num_splits,
-                **partition_kwargs,
-            }
+            rows_skipper = cls.rows_skipper_builder(f, quotechar, is_quoting=is_quoting)
+            rows_skipper(skiprows)
 
             splits = cls.partitioned_file(
                 f,
                 num_partitions=num_partitions,
-                nrows=nrows,
-                skiprows=skiprows,
+                nrows=kwargs.get("nrows", None),
                 quotechar=quotechar,
                 is_quoting=is_quoting,
             )
-            for start, end in splits:
-                args.update({"start": start, "end": end})
-                partition_id = cls.deploy(cls.parse, num_splits + 2, args)
-                partition_ids.append(partition_id[:-2])
-                index_ids.append(partition_id[-2])
-                dtypes_ids.append(partition_id[-1])
-
-        # Compute the index based on a sum of the lengths of each partition (by default)
-        # or based on the column(s) that were requested.
+
+        partition_ids, index_ids, dtypes_ids = cls._launch_tasks(
+            splits, **partition_kwargs
+        )
+
+        new_index, row_lengths = cls._define_index(
+            index_ids, index_col, empty_pd_df.index.name
+        )
+
+        # Compute dtypes by getting collecting and combining all of the partitions. The
+        # reported dtypes from differing rows can be different based on the inference in
+        # the limited data seen by each worker. We use pandas to compute the exact dtype
+        # over the whole column for each column. The index is set below.
+        dtypes = cls.get_dtypes(dtypes_ids) if len(dtypes_ids) > 0 else None
+        # Compose modin partitions from `partition_ids`
+        partition_ids = cls.build_partition(partition_ids, row_lengths, column_widths)
+        column_names, column_widths = cls._define_column_names(
+            column_names, parse_dates, column_widths
+        )
+
+        # Set the index for the dtypes to the column names
+        if isinstance(dtypes, pandas.Series):
+            dtypes.index = column_names
+        else:
+            dtypes = pandas.Series(dtypes, index=column_names)
+
+        new_frame = cls.frame_cls(
+            partition_ids,
+            new_index,
+            column_names,
+            row_lengths,
+            column_widths,
+            dtypes=dtypes,
+        )
+
+        new_query_compiler = cls._get_new_qc(
+            new_frame,
+            kwargs.get("skipfooter", None),
+            kwargs.get("squeeze", False),
+            index_col,
+        )
```

Comment:
All this can be moved into the function for creating query_compiler.

Reply:
Right, fixed.